packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_document_reference.dart

```diff
@@ -104,32 +105,78 @@ class MethodChannelDocumentReference extends DocumentReferencePlatform {
     // subscribers have cancelled; this analyzer warning is safe to ignore.
     StreamController<DocumentSnapshotPlatform>
         controller; // ignore: close_sinks
-    controller = StreamController<DocumentSnapshotPlatform>.broadcast(
-      onListen: () async {
-        MethodChannelFirebaseFirestore.documentObservers[handle] = controller;
-        await MethodChannelFirebaseFirestore.channel.invokeMethod<void>(
-          'DocumentReference#addSnapshotListener',
-          <String, dynamic>{
-            'handle': handle,
-            'firestore': firestore,
-            'reference': this,
-            'includeMetadataChanges': includeMetadataChanges,
-          },
-        );
 
-        if (!onListenComplete.isCompleted) {
-          onListenComplete.complete();
-        }
-      },
-      onCancel: () async {
-        await onListenComplete.future;
-        await MethodChannelFirebaseFirestore.channel.invokeMethod<void>(
-          'Firestore#removeListener',
-          <String, dynamic>{'handle': handle},
-        );
-        MethodChannelFirebaseFirestore.documentObservers.remove(handle);
-      },
-    );
+    if (Platform.isIOS) {
```

Comment:
Is this not supported on Android?

If it must only be non-Android then a check for macOS should be done as well as this plugin works with macOS also through the same iOS codebase.

(same for other platform checks in other files)

Reply:
This will be supported on Android once I write the code for it.  ;-)

Thank you for the comment rgd. macOS, will fix!

I'm not sure how _web_ is doing this, yet - just a matter of getting to it, though.

Reply:
> This will be supported on Android once I write the code for it. ;-)

Nice, thanks :D

> I'm not sure how web is doing this, yet - just a matter of getting to it, though.

Should be a non-issue on web as web doesn't use a method channel, calls the JS SDKs directly. So does not go through these method channel related files (purely for mobile & desktop).

Reply:
Note there's a web CI issue due to new nullsafety beta which I'm fixing currently in #4210 - will fail until thats merged and synced into this branch

Reply:
OK got it. 👍